### ISSUE TITLE
Incompatibility with PaperWM extension

### ISSUE THREAD

**Issue by lambdaclan** [flexagoon/rounded-window-corners#9]
Hello there!  Thank you for reviving this extension to work with latest GNOME!  For people who want a consistent look of all their applications rounded corners is a key extension to use with older GTK3 apps along adw-gtk3 theme since libadwaita applications default to rounded corners.

Now, I know this feature request might be low priority since it is not really an issue with this extension but more of a conflict with a different extension. Saying that for all of us that use [PaperWM](https://github.com/paperwm/PaperWM) as a form of tiling within GNOME environment would appreciate it if this extension could add compatibility with it. I noticed that an effort was made to ensure this extension is compatible with blur my shell extension, so I guess it is worth asking for support with yet another conflicting extension ;)

I am not sure how much work it entails or if changes need to be made on this extension or PaperWM side but some sort of collaboration between the extension developers will be possibly necessary.

There was a similar feature request on the old unmaintained repository [`Visual glitches when used with paperWM`](https://github.com/yilozt/rounded-window-corners/issues/154) and there is an open issue on the PaperWM repository [ Incompatibility with rounded corners extension](https://github.com/paperwm/PaperWM/issues/763).

The PaperWM developers seem to be open to PRs to resolve this issue and I assume will be willing to offer advice if needed.

One weird thing I noticed while using the extensions together is that electron based apps, although they are still using GTK3 do not produce the same artifacts as native pure GTK3 apps.

To reproduce the issue

1. Install rounded-window-corners extension with default settings
2. Install paperwm extension with default settings
3. Install any [libadwaita app](https://arewelibadwaitayet.com/)
4. Install any GTK3 app, something like gitg, gnome disks, dconf editor etc
5. Launch the two apps and look at the artifacts on the GTK3 app vs the libadwaita one
6. Optional: You can also try an electron app to verify my findings above.

Once more thank you for forking and maintaining this extension and your consideration for this feature request!

**Comment by jonseppanen** [comment]
Another related issue (on xorg, as this all mostly works on xorg!), is that when PaperWM does its magic to move windows around, the shader effect breaks a tiny but just on the corners. Its like one of the last passes of the cornering needs a refresh or something, cant really explain it.

**Comment by lambdaclan** [comment]
PaperWM had a bug where even if you had set the border selection size to zero it would still display a small section of it. That was making rounded corners behave even worse but it was recently fixed https://github.com/paperwm/PaperWM/pull/861 so now incompatible windows look like below:

![image](https://github.com/flexagoon/rounded-window-corners/assets/47409392/c685b960-0a6c-4825-b05d-09efc6fa5e28)

The window is correctly rounded but there is some weird outline still visible, also the top side parts get cut off.

If you increase border selection size it gets worse but it might be a good way to debug it when developing a workaround:

![image](https://github.com/flexagoon/rounded-window-corners/assets/47409392/f513d779-3901-4f7f-b5a0-74f9edc412d3)

Something similar to the issue mentioned by @jonseppanen this is how Chromium looks upon launch:

![image](https://github.com/flexagoon/rounded-window-corners/assets/47409392/4f739019-287b-40c1-a0d1-052dc9dcbfa2)

but if you cause a refresh for example toggle full screen on/off then it looks correct:

![image](https://github.com/flexagoon/rounded-window-corners/assets/47409392/d70f147f-e835-45d7-b2ac-cbb8b34f709b)

Hopefully this info helps out a bit!

**Comment by Sepax** [comment]
got same issue

**Comment by sound-friend** [comment]
any updates to this?

**Comment by flexagoon** [comment]
@sound-friend nope, unfortunately, this is currently not a priority issue for me as I don't use PaperWM, sorry. I'd appreciate if someone could help with investigating the cause of this though

**Comment by vaibhavs-mishra** [comment]
With PaperWM enabled, there are glitches with Firefox based browsers, where the rounded corners work but the window is cropped into and the cropped portion is invisible. Screenshots don't show it.

Using Wayland. 
I will remove paperwm for now. I only needed the window rules functionality, now to find another extension for it.
